### ISSUE TITLE
rest doesn't work with object destructuring

### ISSUE THREAD

**Issue by AsaAyers** [eslint/espree#131]
I found this because eslint is reporting the error, but I'm fairly sure the bug is really here in espree.

```Javascript
const foo = {a: 1, b: 2, c: 3}
const { a, ...bar } = foo;
//         ^ Unexpected token ...

console.log(a); // 1
console.log(bar) // {"b":2, "c":3}
```
You can see this code in Babel's [repl](https://babeljs.io/repl/#?experimental=true&evaluate=true&loose=false&spec=false&code=const%20foo%20%3D%20%5B1%2C%202%2C%203%5D%0Aconst%20%5B%20a%2C%20...bar%20%5D%20%3D%20foo%3B%0A%0Aconsole.log(a)%3B%20%2F%2F%201%0Aconsole.log(bar)%20%2F%2F%20%5B2%2C3%5D).

It seems like it's fine when destructuring arrays.

```
const foo = [1, 2, 3]
const [ a, ...bar ] = foo;

console.log(a); // 1
console.log(bar) // [2,3]
```

**Comment by lo1tuma** [comment]
You are using the [rest properties operator](https://github.com/sebmarkbage/ecmascript-rest-spread) which is a proposal for ES7.

**Comment by AsaAyers** [comment]
You're right. I didn't realize that "Experimental" is checked by default on Babel's repl.

**Comment by AsaAyers** [comment]
For what it's worth, I find ESLint to be so valuable that I won't actually commit code it doesn't work with. I started using most of ES6 but avoided classes until they were added a few releases ago.

Now I'm just curious. I saw on http://babeljs.io/blog/2015/03/31/5.0.0/ that TC-39 has maturity stages. Have there been any plans to have espree support everything with a maturity level over a specific level?

**Comment by nzakas** [comment]
Espree will only support standard features - I just don't have the time to commit to implementing every new experiment. If you're using ESLint, then I'd suggest using babel-eslint as the parser to get all the experimental features you want.

**Comment by AsaAyers** [comment]
I don't want experimental features. I was just not aware it was experimental.
